addons/hr_recruitment/models/hr_job.py

```diff
@@ -186,6 +188,16 @@ def _compute_application_count(self):
         for job in self:
             job.application_count = result.get(job.id, 0)
 
+    def _compute_open_application_count(self):
+        hired_stages = self.env['hr.recruitment.stage'].search([('hired_stage', '=', True)])
+        open_applications_data = self.env['hr.applicant']._read_group([
+            ('job_id', 'in', self.ids),
+            ('stage_id', 'not in', hired_stages.ids),
+        ], ['job_id'], ['__count'])
+        result = {job.id: count for job, count in open_applications_data}
+        for job in self:
+            job.open_application_count = result.get(job.id, 0)
```

Comment:
```suggestion
        open_applications_data = dict(self.env['hr.applicant']._read_group([
            ('job_id', 'in', self.ids),
            ('stage_id', 'not in', hired_stages.ids),
        ], ['job_id'], ['__count']))
        for job in self:
            job.open_application_count = result.get(job, 0)
```
This should work I think :)